Add gruvbox base theme and gruvbox green color theme

DeaDvey opened the pull request:
#  Add gruvbox base theme and gruvbox green color theme

## Pull Request Type
<!-- Please select what type of pull request this is: [x] -->

- [x] Feature Implementation

## Related issue
Closes #3987

## Description
I added a gruvbox base theme using the gruvbox colour palette as well as a gruvbox green color theme, I will add more of the color themes if the pr is accepted 

## Screenshots <!-- If appropriate -->
![settings page](https://github.com/FreeTubeApp/FreeTube/assets/104674266/a19a76c1-7234-4628-910c-0b1f4d9977e0)
![subscriptions page](https://github.com/FreeTubeApp/FreeTube/assets/104674266/b662f908-418d-4e8e-9f16-93a18fdff589)
![watch page](https://github.com/FreeTubeApp/FreeTube/assets/104674266/93992c09-60ed-4304-b8da-2dcec826fb38)

## Testing <!-- for code that is not small enough to be easily understandable -->
I tested it and I cannot see any issues with the color schemes showing though I have not added all the translations as I only speak english.

## Desktop
<!-- Please complete the following information-->
- **OS:** GNU/Linux
- **OS Version:** openSUSE Tumbleweed
- **FreeTube version:** most up to date one on github

## Additional context
I wanted a gruvbox theme for FreeTube and I saw that another issue request did too, it's fine if you (the developers) don't, I was just opening this pr incase it was appropriate.

DeaDvey commented:
Ah, lint.

absidue reviewed (commented):
Thank you for opening your first pull request on the FreeTube repository 🥳.

Please fix the linter errors, if you switch to the files changed tab, you can see all the alerts.

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (changes requested):
Stuff disappears when both primary and secondary colors are set to `gruvbox green`

https://github.com/FreeTubeApp/FreeTube/assets/73130443/5c92bc68-9804-4f87-84e2-35fd05c8a225

efb4f5ff-1298-471a-8973-3d47447115dc commented:
Also this doesnt fully close the issue because you didnt implement all the gruvbox colors listed in their repo https://github.com/morhetz/gruvbox. The issue also explicitly asks for the light and dark theme implementation.

DeaDvey commented:
Yes, I will implement both light and dark, I was just checking if this pr is appropriate. I'll fix the secondary colour issue right now,
Thanks

DeaDvey commented:
Ok, so the reason the secondary colours were not working properaly was because I did not add the "--accent-color-opacity1 - 4" so I added that and I think it's fixed though I haven't added the gruv light theme yet.

DeaDvey commented:
By the way, I'm not sure how to make the icons for it so mine were a bit thrown together and look not great, so yeah.

DeaDvey commented:
![image](https://github.com/FreeTubeApp/FreeTube/assets/104674266/01707782-c500-4a40-ad65-744bb1f2b7bf)
![image](https://github.com/FreeTubeApp/FreeTube/assets/104674266/4dccd0e0-bc48-4741-b775-52d655c3f454)
![image](https://github.com/FreeTubeApp/FreeTube/assets/104674266/cffdefa5-054b-452c-91dc-ac7e99c299f9)

So I added all the colour themes as well @efb4f5ff-1298-471a-8973-3d47447115dc
I tested the themes and cannot see any issues (I fixed the secondary colour scheme issue you raised)
Though I still have not added translations and the logos look bad as I don't know how to make them for the theme. 
![image](https://github.com/FreeTubeApp/FreeTube/assets/104674266/c08e9315-e2a0-4d2a-a6cc-b6599b377468)

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> So I added all the colour themes as well @efb4f5ff-1298-471a-8973-3d47447115dc I tested the themes and cannot see any issues (I fixed the secondary colour scheme issue you raised) Though I still have not added translations 

Can confirm that the issue is fixed and cant find any other issues at the moment :) 
You dont have to add the translations for `en_Gb` as the translators will do that for you.  

> the logos look bad as I don't know how to make them for the theme. ![image](https://private-user-images.githubusercontent.com/104674266/316687844-c08e9315-e2a0-4d2a-a6cc-b6599b377468.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE1NzUwOTMsIm5iZiI6MTcxMTU3NDc5MywicGF0aCI6Ii8xMDQ2NzQyNjYvMzE2Njg3ODQ0LWMwOGU5MzE1LWUyYTAtNGQyYS1hNmNjLWI2NTk5YjM3NzQ2OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyN1QyMTI2MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04N2RkZTNjNDI0MWI0NmNhMDRjN2ZhMTM5NTkwNzI3YmZlM2FiMGRiODAwMjIxN2UzNDgxYTdjNjVjMDkxY2Q3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.FN7Dx6L1bpNlvRt6N-giCedCVjXLHD6kWhZ0lFmOtOU)

Maybe previous contributors can give you some guidance on that. Summoning previous contributors that created PR's for adding themes:
@codysechelski
@Rekit0
@JonahFoster

JonahFoster commented:
@DeaDvey For the logos, what I did was upload it to [Photopea ](https://www.photopea.com/)and just replace the colors on the main logo. You can find a ton of videos and blogs online about photopea, and here are some [Reddit comments](https://www.reddit.com/r/photopea/comments/bzfoem/replacing_colours_in_an_image/) that might be helpful. Feel free to reach out if you have any specific questions

DeaDvey commented:
Thanks, I'll check out Photopea, I think gimp worked for me here though, for future note, I used the colorise tool to change the colour of the whole thing to #ebdbb2 and #282828

DeaDvey commented:
And I realised the en_GB was not needed, I added that first as that's the one I thought to add first (I'm british) but realised en-US would be better.

efb4f5ff-1298-471a-8973-3d47447115dc commented:
When #4898 will be merged you have some higher quality icons available. I would recommend to redo your icons based on those enhanced ones

DeaDvey commented:
Not ready right now, one of the logos is showing as dark in dark mode and the main and secondary themes don't seem to be showing, can't work on it until tomorrow though.  I did implement SVG images though which seem to work.

DeaDvey commented:
Looks good to me, can't see any issues with the svgs loading and they're the correct ones.

jasonhenriquez commented on src/renderer/themes.css:
```diff
@@ -261,6 +271,45 @@ it can be safely elided. This looks quite pleasant on this theme. */
+  --favorite-icon-color: #fabd2f;
+  --card-bg-color: #282828;
+  --secondary-card-bg-color: #282828;
+  --scrollbar-color: #32302f;
```
**issue (blocking):** Please choose a color that has [an acceptable color contrast](https://webaim.org/resources/contrastchecker/) (>=4.5:1) with the background (`--bg-color`). This scrollbar can be hard to find even for non-colorblind users.

jasonhenriquez commented on src/renderer/themes.css:
```diff
@@ -261,6 +271,45 @@ it can be safely elided. This looks quite pleasant on this theme. */
+  --bg-color: #1d2021;
+  --favorite-icon-color: #fabd2f;
+  --card-bg-color: #282828;
+  --secondary-card-bg-color: #282828;
```
**issue (blocking):** Please choose a color that has an acceptable color contrast (>=3:1) with the card background color (`--card-bg-color`). Examples of where this is used: the "About" tags on a channel, the author name outline on a comment made by the video author in a video's comments section.

jasonhenriquez reviewed (changes requested):
**issue (blocking):** You'll also need to modify the [ft-share-button SCSS](https://github.com/jasonhenriquez/FreeTube/blob/development/src/renderer/components/ft-share-button/ft-share-button.scss) (note to team: we should honestly centralize this in themes.css in the future as this is very easy to miss).

DeaDvey commented:
Hi, I fixed the colour contrast issues mentioned but I wasn't 100% sure on the ft-share-button thing, I think I did what you requested but you'll have to check.
I also noticed a conflicting file so updated mine to incorporate the changes but I don't have write access to resolve the conflict.

jasonhenriquez commented on src/renderer/components/ft-share-button/ft-share-button.scss:
```diff
@@ -53,6 +54,7 @@
 
       .pastelPink &,
       .light &,
+      .gruvboxLIght &,
```
```suggestion
      .gruvboxLight &,
```

jasonhenriquez commented:
You can resolve the conflict with a merge or rebase with the `development` branch.

jasonhenriquez commented:
> Hi, I fixed the colour contrast issues mentioned but I wasn't 100% sure on the ft-share-button thing, I think I did what you requested but you'll have to check.

What this governs is what version of the Invidious and YouTube logos show when you click the "Share" icon under a video. If the logo is too dark / bright to see, it means it needs to go in with either the dark or light colors. I haven't checked on this yet, but from a brief scan at the file, it seems like you got it.

jasonhenriquez commented:
From what I can see, yes! As a reminder in case my message above was missed:
> You can resolve the conflict with a merge or rebase with the development branch.

DeaDvey commented:
Hmmm, Lint says there's no issues on my machine.

absidue reviewed (changes requested):
Please remove the `package-lock.json` file (FreeTube uses yarn) and undo the addition of the `lint` dependency.

jasonhenriquez commented:
@DeaDvey Could you please rebase/merge with development one more time and remove the changes to `yarn.lock`?

DeaDvey commented:
Sorry that I took a while, I'm quite busy at the moment.

DeaDvey commented:
I get an error saying there is no portal-vue. in node_modules

absidue commented:
Run `yarn install` to download the dependency that you are missing.

DeaDvey commented:
Oh yeah, thanks that worked, the themes all show correctly, solarized looks great too! 👍

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (changes requested):
Fix poo contrast issues:

- https://coolors.co/contrast-checker/fb4934-282828
- https://coolors.co/contrast-checker/b8bb26-fbf1c7
- https://coolors.co/contrast-checker/fb4934-fbf1c7
- https://coolors.co/contrast-checker/fabd2f-fbf1c7
- https://coolors.co/contrast-checker/83a598-fbf1c7
- https://coolors.co/contrast-checker/d3869b-fbf1c7
- https://coolors.co/contrast-checker/8ec07c-fbf1c7
- https://coolors.co/contrast-checker/fe8019-fbf1c7

jasonhenriquez commented:
Thanks for identifying these @efb4f5ff-1298-471a-8973-3d47447115dc. Seems like `#fb1c7` should be changed to something else with a higher color contrast, from the looks of it. That and maybe having a separate `bg-color` and `card-bg-color` for `gruvboxLight`.

jasonhenriquez commented:
Hi @DeaDvey, just checking in to make sure you're alright. I want to make sure the required suggestion and how to implement & validate it are clear enough.

DeaDvey commented:
Yeah, I'm a bit busy right now but I should be able to have a look at it tomorrow.

DeaDvey commented:
I think in order to fix this I'd either need to add a  whole new set of Color Themes for Gruvbox Light as they use different greens, reds, yellows etc or remove the Gruvbox Light base theme all together. As the ones I implemented  only have a good colour contrast with Gruvbox Dark themes.

https://github.com/morhetz/gruvbox?tab=readme-ov-file

DontBlameMe99 commented:
@DeaDvey Couldn't you just create gruvbox light green / gruvbox dark green respectively? Or would this be too much of a spam of colors?

DeaDvey commented:
> @DeaDvey Couldn't you just create gruvbox light green / gruvbox dark green respectively? Or would this be too much of a spam of colors?

Yeah, that's what I was thinking, it would be too many colours for the menu to look nice, I could definitley do it but I don't think it's a good idea.

jasonhenriquez commented:
Hi @DeaDvey, I agree that either removing the new primary/secondary color themes or removing one of the base themes (and adjusting the new primary/secondary colors in the case that Gruvbox Light is the one that's kept) should work.

DeaDvey commented:
Ok thanks, jasonhenriquez, I will remove Gruvbox Light as light themes are typically less popular so I think it's thats right choice, this should fix all the mentioned contrast issues.

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> Fix poor contrast issues:
> 
> - https://coolors.co/contrast-checker/fb4934-282828
> - https://coolors.co/contrast-checker/b8bb26-fbf1c7
> - https://coolors.co/contrast-checker/fb4934-fbf1c7
> - https://coolors.co/contrast-checker/fabd2f-fbf1c7
> - https://coolors.co/contrast-checker/83a598-fbf1c7
> - https://coolors.co/contrast-checker/d3869b-fbf1c7
> - https://coolors.co/contrast-checker/8ec07c-fbf1c7
> - https://coolors.co/contrast-checker/fe8019-fbf1c7
> 
> @jasonhenriquez do you have suggestions to handle this in the best possible way?

It doesnt fix the first issue in the list

DeaDvey commented:
> It doesnt fix the first issue in the list
Sorry, yeah, I changed the central pane to be #1d2021 so the contrast with #fb4934 is 4.77 (good) https://coolors.co/contrast-checker/fb4934-1d2021

efb4f5ff-1298-471a-8973-3d47447115dc commented on static/locales/en_GB.yaml:
```diff
@@ -288,6 +288,7 @@ Settings:
       Pastel Pink: Pastel pink
       Hot Pink: Hot pink
       Nordic: Nordic
+      Gruvbox: Gruvbox
```
This should state the same as https://github.com/FreeTubeApp/FreeTube/pull/4805/files#diff-4eaadcc25bfbd7ab1ba912e36d5e1bcbab346cab59546a970df7546d5fb7a145R331

```suggestion
      Gruvbox Dark: Gruvbox Dark
```

efb4f5ff-1298-471a-8973-3d47447115dc commented on src/renderer/themes.css:
```diff
@@ -461,6 +541,8 @@ it can be safely elided. This looks quite pleasant on this theme. */
   --logo-text-bar-color: url("../../_icons/textDraculaLightSmall.svg");
 }
 
+
```
```suggestion
```

efb4f5ff-1298-471a-8973-3d47447115dc commented on src/renderer/themes.css:
```diff
@@ -461,6 +541,8 @@ it can be safely elided. This looks quite pleasant on this theme. */
 }
 
+
+
```
```suggestion
```

efb4f5ff-1298-471a-8973-3d47447115dc commented on src/renderer/themes.css:
```diff
@@ -444,6 +468,62 @@ it can be safely elided. This looks quite pleasant on this theme. */
+    --primary-color-active: #af3a03;
+}
+
+
```
```suggestion
```

efb4f5ff-1298-471a-8973-3d47447115dc commented on src/renderer/components/theme-settings/theme-settings.js:
```diff
@@ -39,6 +39,7 @@ export default defineComponent({
         'black',
         // Second group
         'nordic',
+        'gruvboxDark',
```
Please insert this in the third group below `dracula`
https://github.com/FreeTubeApp/FreeTube/pull/4805/files#diff-a0480aa13261014e61fefded30097805ea7559cfe7b068b66b209ae896b17adaR47
```suggestion
```

efb4f5ff-1298-471a-8973-3d47447115dc commented on src/renderer/components/theme-settings/theme-settings.js:
```diff
@@ -111,6 +112,7 @@ export default defineComponent({
         this.$t('Settings.Theme Settings.Base Theme.Black'),
         // Second group
         this.$t('Settings.Theme Settings.Base Theme.Nordic'),
+        this.$t('Settings.Theme Settings.Base Theme.Gruvbox Dark'),
```
Insert this below 
https://github.com/FreeTubeApp/FreeTube/pull/4805/files#diff-a0480aa13261014e61fefded30097805ea7559cfe7b068b66b209ae896b17adaR120
```suggestion
```

efb4f5ff-1298-471a-8973-3d47447115dc commented on src/renderer/helpers/colors.js:
```diff
@@ -31,6 +31,13 @@ export const colors = [
   { name: 'CatppuccinMochaSapphire', value: '#74C7EC' },
   { name: 'CatppuccinMochaBlue', value: '#89B4FA' },
   { name: 'CatppuccinMochaLavender', value: '#B4BEFE' },
+  { name: 'GruvboxGreen', value: '#b8bb26' },
```
put this below the dracula ones

efb4f5ff-1298-471a-8973-3d47447115dc commented on src/renderer/helpers/colors.js:
```diff
@@ -80,6 +87,13 @@ export function getColorTranslations() {
     i18n.t('Settings.Theme Settings.Main Color Theme.Catppuccin Mocha Sapphire'),
     i18n.t('Settings.Theme Settings.Main Color Theme.Catppuccin Mocha Blue'),
     i18n.t('Settings.Theme Settings.Main Color Theme.Catppuccin Mocha Lavender'),
+    i18n.t('Settings.Theme Settings.Main Color Theme.Gruvbox Green'),
```
put these colors below the dracula ones

jasonhenriquez commented on src/main/index.js:
```diff
@@ -626,6 +626,8 @@ function runApp() {
         case 'nordic':
           return '#2b2f3a'
+        case 'gruvbox-dark':
+          return '282828'
```
This needs to be updated to the new bg-color of `32302f`.

jasonhenriquez commented on src/renderer/components/ft-share-button/ft-share-button.scss:
```diff
@@ -56,6 +57,7 @@
 
       .pastelPink &,
       .light &,
+      .gruvboxLight &,
```
```suggestion
      .light &,
```

Remove extraneous reference to gruvBoxLight

jasonhenriquez commented on src/renderer/components/ft-share-button/ft-share-button.scss:
```diff
@@ -79,13 +81,15 @@
         }
 
         .pastelPink &,
         .light &,
+	.gruvboxLight &,
```
```suggestion
        .light &,
```

Remove extraneous reference to gruvBoxLight

jasonhenriquez commented on src/renderer/themes.css:
```diff
@@ -444,6 +468,61 @@ it can be safely elided. This looks quite pleasant on this theme. */
+.mainGruvboxPurple,
+.mainGruvboxAqua,
+.mainGruvboxOrange {
+    --text-with-main-color: #282828;    
```
This `text-with-main-color`, and the other `--text-with-accent-color`, (which should also both be the same color) are still at a notably low color contrast with `mainGruvboxRed`  and `secGruvboxRed`(4.29:1 and 4.77:1 respectively). Ideally this should be a 7:1 contrast to meet [the WCAG AAA standard](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html).

efb4f5ff-1298-471a-8973-3d47447115dc commented:
@DeaDvey some code reviews are still unresolved/not implemented
